zipkin-lens/src/components/TracePage/SpanDetail/SpanAnnotations.jsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import React, {
+  useState,
+  useEffect,
+  useCallback,
+  useMemo,
+} from 'react';
+import Box from '@material-ui/core/Box';
+import Button from '@material-ui/core/Button';
+
+import { generateAnnotationKey } from './util';
+import SpanAnnotation from './SpanAnnotation';
+import SpanAnnotationGraph from './SpanAnnotationGraph';
+import { detailedSpanPropTypes } from '../../../prop-types';
+
+const propTypes = {
+  span: detailedSpanPropTypes.isRequired,
+};
+
+const SpanAnnotations = ({ span }) => {
+  const [currentAnnotationKey, setCurrentAnnotationKey] = useState();
+  const [areAllAnnotationsOpened, setAreAllAnnotationsOpened] = useState(false);
+
+  const handleAnnotationCircleClick = useCallback((annotation) => {
+    // If all annotations have already been expanded, close all of them.
+    if (areAllAnnotationsOpened) {
+      setAreAllAnnotationsOpened(false);
+    }
+    // If an annotation that has already been focused is selected, unfocus it.
+    const key = generateAnnotationKey(annotation);
+    setCurrentAnnotationKey(prev => (prev === key ? null : key));
+  }, [areAllAnnotationsOpened]);
+
+  // Initialize states when a different span is selected.
+  useEffect(() => {
+    setCurrentAnnotationKey(null);
+    setAreAllAnnotationsOpened(false);
+  }, [span.spanId]);
+
+  const currentAnnotation = useMemo(() => span.annotations.find((annotation) => {
```

Comment:
This is `true` or `false`, so a name like `isCurrentAnnotation` or `currentAnnotationOpened` would be easier to follow.

Reply:
Oh, `currentAnnotation` is not boolean value.

Reply:
Ah oops, that was just the argument to `find` - got confused